src/coreclr/jit/morph.cpp

```diff
@@ -8916,7 +8916,7 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
 
             noway_assert(op1);
 
-            if (op1->OperIsCompare())
+            if (op1->OperIsCompare() || op1->OperIs(GT_AND))
```

Comment:
Do we actually need this special handling in morph? What does it accomplish?

Reply:
Intention was to ensure the AND chain got generated into flags instead of registers. Looking at an spmidiff now, it doesn't seem to be doing anything. Removed.